flang/lib/Optimizer/Dialect/FIROps.cpp

```diff
@@ -1787,6 +1843,10 @@ llvm::LogicalResult fir::EmboxOp::verify() {
     return emitOpError("slice must not be provided for a scalar");
   if (getSourceBox() && !mlir::isa<fir::ClassType>(getResult().getType()))
     return emitOpError("source_box must be used with fir.class result type");
+  if (fir::isa_volatile_type(getMemref().getType()) !=
+      fir::isa_volatile_type(getResult().getType()))
+    return emitOpError("cannot convert between volatile and non-volatile "
+                       "types, use fir.volatile_cast instead");
```

Comment:
I believe fir.convert is used for PTR to INT conversions. Isn't this causing verifier issues when using LOC on a VOLATILE variable?

Reply:
Yes, I just found this test in the gfortran test suite that shows this:
```
llvm-test-suite/Fortran/gfortran/regression/dec_loc_rval_2.f90
```
I'll correct it, thank you!